packages/astro/src/env/schema.ts

```diff
@@ -75,11 +75,28 @@ const SecretServerEnvFieldMetadata = z.object({
 	context: z.literal('server'),
 	access: z.literal('secret'),
 });
-const EnvFieldMetadata = z.union([
+const _EnvFieldMetadata = z.union([
 	PublicClientEnvFieldMetadata,
 	PublicServerEnvFieldMetadata,
 	SecretServerEnvFieldMetadata,
 ]);
+const EnvFieldMetadata = z.custom<z.input<typeof _EnvFieldMetadata>>().superRefine((data, ctx) => {
+	const result = _EnvFieldMetadata.safeParse(data);
+	if (result.success) {
+		return;
+	}
+	for (const issue of result.error.issues) {
+		if (issue.code === z.ZodIssueCode.invalid_union) {
+			ctx.addIssue({
+				code: z.ZodIssueCode.custom,
+				message: `**Invalid combination** of "access" and "context" options:\n  Review the configuration of \`env.schema.${ctx.path.at(-1)}\`.\n  Learn more at https://docs.astro.build/en/guides/environment-variables/#variable-types`,
```

Comment:
```suggestion
				message: `**Invalid combination** of "access" and "context" options:\n  Review the configuration of \`env.schema.${ctx.path.at(-1)}\`.\n  Learn more at https://docs.astro.build/en/guides/environment-variables/#variable-types`,
```
Some questions!

- Do error messages usually have some bolded words? This jumps out to me as odd.
- If I remember correctly, isn't there only *one* invalid combination? So, instead of saying "invalid combination", is it possible to describe that "You can't have secret variables that are available on the client" or something like that?

Reply:
> Do error messages usually have some bolded words?

This is only allowed by zod I believe, and I just thought it was nice since zod uses bold errors for its default error messages. Happy not to use that tho!

> If I remember correctly, isn't there only one invalid combination

Ah yes you're right! I'll update the message

---

Btw you sent a suggestion but I think you didn't update anything, idk if you forgot something

Reply:
```suggestion
				message: `**Invalid combination** of "access" and "context" options:\n  Server client variables are not supported, review the configuration of \`env.schema.${ctx.path.at(-1)}\`.\n  Learn more at https://docs.astro.build/en/guides/environment-variables/#variable-types`,
```

Reply:
@florian-lefebvre I updated, but didn't commit because I *think* you meant "Secret client variables" (not "Server client variables" because this is about a mismatch between access and context?). But, since you also can't have "Server client" variables, (it's not wrong! 😅 ) I want to make sure you're proofing this!

Reply:
Those 2 words are so close, you better believe I made a mistake